ncard_app/views.py

```diff
@@ -1,10 +1,39 @@
-from django.shortcuts import render
+from django.shortcuts import render, redirect
+from django.contrib.auth import authenticate, login, logout
 from ncard_app.models import Award
+from django.contrib import messages
+
 def index(request):
     template_context = {
         'value': 123,
     }
-    return render(request, 'index.html', template_context)
+    return render(request, 'events/index.html', template_context)
+
+def home(request):
+    if not request.user.is_authenticated:
+        messages.error(request, ("Please login to access this page."))
+        return redirect('login')
+    return render(request, 'events/home.html', {})
+
+def login_user(request):
+    if request.method == "POST":
+        username = request.POST['username']
+        password = request.POST['password']
+        user = authenticate(request, username=username, password=password)
+        if user is not None:
+            login(request, user)
+            return redirect('home')
+        else:
+            messages.error(request, ("There was an error logging in."))
```

Comment:
This message should be something like "Incorrect username or password." Otherwise, it sounds like the server is having a problem and the user can't fix it.